kotlin/src/main/com/looker/rtl/ApiSettings.kt

```diff
@@ -55,7 +57,7 @@ open class ApiSettingsIniFile(var filename: String = "./looker.ini",
 
 
 // TODO why no @JvmOverloads here?
-open class ApiSettings(contents: String, var section: String = "") : ConfigurationProvider {
+open class ApiSettingsIniText(contents: String, var section: String = "") : ApiSettings, ConfigurationProvider {
```

Comment:
I wanted to combine this into the ApiSettingsIniFile, perhaps as a named static constructor (companion object method in kotlin), but they both had `readConfig()` implementations, so I didn't want to rock the boat too much by reconciling those.

Reply:
the idea behind `readConfig()` is that it's supposed to support overriding from the base so you can get your configuration any way you want. As long as you're not breaking that "inherit and override" capability, changes should be fine.

Reply:
My intent with `ApiSettings` was for it to be the base overrideable "config reader" implementation that happened to default to `.ini` text, and other inherited overrides would be more explicitly named. I'm wondering how many people would see `ApiSettingsIniText` or `ApiSettingsIniFile` and think "Oh, that's not the one I want to descend from."